nautobot_chatops/sockets/slack.py

```diff
@@ -0,0 +1,216 @@
+"""Websocket Client to receive inbound notifications from Slack, parse them, and enqueue worker actions."""
+
+import asyncio
+import json
+import shlex
+
+from django.conf import settings
+from slack_sdk.socket_mode.aiohttp import SocketModeClient
+from slack_sdk.socket_mode.response import SocketModeResponse
+from slack_sdk.socket_mode.request import SocketModeRequest
+from slack_sdk.web.async_client import AsyncWebClient
+
+from nautobot_chatops.workers import get_commands_registry, commands_help, parse_command_string
+from nautobot_chatops.dispatchers.slack import SlackSocketDispatcher
+from nautobot_chatops.utils import socket_check_and_enqueue_command
+
+
+SLASH_PREFIX = settings.PLUGINS_CONFIG["nautobot_chatops"].get("slack_slash_command_prefix")
+
+
+async def main():  # pylint: disable=too-many-statements
+    """Slack Socket Main Loop."""
+    client = SocketModeClient(
+        app_token=settings.PLUGINS_CONFIG["nautobot_chatops"].get("slack_app_token"),
+        web_client=AsyncWebClient(token=settings.PLUGINS_CONFIG["nautobot_chatops"]["slack_api_token"]),
+    )
+
+    async def process(client: SocketModeClient, req: SocketModeRequest):
+        client.logger.debug("SK Received a socket request.")
+        if req.type == "slash_commands":
+            client.logger.debug("Received slash command.")
+            # Acknowledge the request anyway
+            response = SocketModeResponse(envelope_id=req.envelope_id)
+            # Don't forget having await for method calls
+            await client.send_socket_mode_response(response)
+            await process_slash_command(client, req)
+
+        if req.type == "interactive":
+            client.logger.debug("Received interactive request.")
+            # Acknowledge the request anyway
+            response = SocketModeResponse(envelope_id=req.envelope_id)
+            # Don't forget having await for method calls
+            await client.send_socket_mode_response(response)
+            await process_interactive(client, req)
+
+    async def process_slash_command(client, req):
+        client.logger.debug("Processing slash command.")
+        command = req.payload.get("command")
+        command = command.replace(SLASH_PREFIX, "")
+        params = req.payload.get("text")
+        context = {
+            "org_id": req.payload.get("team_id"),
+            "org_name": req.payload.get("team_domain"),
+            "channel_id": req.payload.get("channel_id"),
+            "channel_name": req.payload.get("channel_name"),
+            "user_id": req.payload.get("user_id"),
+            "user_name": req.payload.get("user_name"),
+            "response_url": req.payload.get("response_url"),
+            "trigger_id": req.payload.get("trigger_id"),
+        }
+
+        try:
+            command, subcommand, params = parse_command_string(f"{command} {params}")
+        except ValueError as err:
+            client.logger.error("%s", err)
```

Comment:
Do we need a `return` here, or is it safe to continue to the rest of the function?

Reply:
Probably a return, I will need to double check how the SlackSlashCommandView handles this.